### DISCUSSION TITLE
Any way to improve text quality / antialiasing?

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4618] [Help] [answered]
Is there any way to make a text look smoother?
I'm using a -pointsize 80 font with a stroke yet my text still looks quite choppy. Certain fonts just plain look bad:

Command:
```
magick input.jpg ^
( -background none -fill #947a27 -stroke #947a27 -strokewidth 10 -font Sylfaen ^
-pointsize 80 -size 550x -gravity center caption:"Marlingbaits" -blur 0x6 ) ^
-gravity North -geometry +0+15 -composite ^
( -background none -fill #fdf9dc -stroke #fdf9dc -strokewidth 1 -font Sylfaen ^
-pointsize 80 -size 550x -gravity center caption:"Marlingbaits" -antialias ) ^
-gravity North -geometry +0+19 -composite ^
result.jpg 
```

![result_maxresdefault (1)](https://user-images.githubusercontent.com/95970582/146722435-55985717-497e-43fb-b31d-8ae7566f76b2.jpg)

![result_maxresdefault (1) - Copy](https://user-images.githubusercontent.com/95970582/146722365-4d4353c8-a199-4377-b117-15feb8ee95b7.jpg)

input image:

![result_maxresdefault (1)](https://user-images.githubusercontent.com/95970582/146723277-df6efab8-1053-4211-b6d0-c76895f34ad3.jpg)

**Comment by snibgo** [comment, marked as answer]
You have "-antialias" _after_ you have drawn the caption, which is too late because it has already been drawn. But "-antialias" enables antialiasing, which is the default, so it would have no effect anyway. Try "+antialias" _before_ "caption:" to see the impact.

I don't see a major problem. You might prefer to draw the text at twice the size, and reduce to half. So the second text would be:
```
( -background none -fill #fdf9dc -stroke #fdf9dc -strokewidth 1 -font Sylfaen ^
-pointsize 160 -size 1100x -gravity center caption:"Marlingbaits" -resize 50% ) ^
```
(The first text is blurred, so not worth the supersampling.)

Another possibility is to draw in linear RGB space instead of non-linear sRGB. After reading the input, insert:
```
-colorspace RGB
```
Before writing the output, convert back to sRGB:
```
-colorspace sRGB
```
You will need to adjust your colours #947a27 etc.

EDIT: And I suggest using "-stroke None" for the main text. That adds an outline to the text, which I think you don't want.

**Reply by SentientBeingOnAQuest** [reply]
-resize Works perfectly! The difference is night and day:

Without -resize:

![result bad](https://user-images.githubusercontent.com/95970582/146751213-7f34501e-7dc4-45f2-91f5-dcb00026b680.jpg)

With -resize:

![result](https://user-images.githubusercontent.com/95970582/146751336-11b3fd90-93f7-41cf-83ff-b5d4bd7590d4.jpg)

Particularly on the first "a" you can really see the difference. 

Thanks again for your amazing help!

**Reply by sergeevabc** [reply]
Indeed, `+antialias` before `-annotate` enabled me to put + (plus, cross, whatever) sign upon image without blurry effect.
